composer/loggers/mosaicml_logger.py

```diff
@@ -0,0 +1,174 @@
+                                 "data": data
+                             },
+                             timeout=3)
+    return response
+
+
+class MosaicMLLoggerBackend(RankZeroLoggerBackend, Serializable):
+    """Log to the MosaicML backend.
+
+    Args:
+        job_id (str, optional): The id of the job to write logs for.
+        sweep_id (str, optional): The id of the sweep to write logs for.
+        creds_file (str, optional): A file containing the MosaicML api_key. If not provided
+            will default to the environment variable MOSAIC_API_KEY.
+        flush_every_n_batches (int): Flush the log data buffer every n batches. (default: ``100``)
+        max_logs_in_buffer (int): The maximum number of log entries allowed in the buffer before a
+            forced flush. (default: ``1000``)
+    """
+
+    def __init__(self,
+                 job_id: Optional[str] = None,
+                 sweep_id: Optional[str] = None,
+                 creds_file: Optional[str] = None,
+                 flush_every_n_batches: int = 100,
+                 max_logs_in_buffer: int = 1000) -> None:
+
+        super().__init__()
+        self.skip_logging = False
+        self.job_id = job_id if job_id is not None else os.environ.get(_JOB_ID_ENV, None)
+        if self.job_id is None:
+            self.skip_logging = True
+            warnings.warn("No job_id provided to MosaicLoggerBackend. MosaicML logger will be a no-op.")
+
+        self.sweep_id = sweep_id if sweep_id is not None else os.environ.get(_SWEEP_ID_ENV, None)
+        if self.sweep_id is None:
+            self.skip_logging = True
+            warnings.warn("No sweep_id provided to MosaicLoggerBackend. MosaicML logger will be a no-op.")
+
+        if creds_file:
+            with open(creds_file, 'r') as f:
+                os.environ[_MOSAICML_API_KEY_ENV] = str(f.read())
+
+        if os.environ.get(_MOSAICML_API_KEY_ENV, None) is None:
+            self.skip_logging = True
+            warnings.warn(
+                f"No api_key set for environment variable {_MOSAICML_API_KEY_ENV}. MosaicML logger will be a no-op.")
+
+        self.buffered_data = []
+        self.flush_every_n_batches = flush_every_n_batches
+        self.max_logs_in_buffer = max_logs_in_buffer
+
+        self.queue = Queue()
+        self.thread = Thread(target=self._listen_to_queue)
+
+    def _log_metric(self, epoch: int, step: int, log_level: LogLevel, data: TLogData):
+        del log_level  # unused
+
+        if self.skip_logging:
+            return
+
+        log_data = {
+            "job_id": self.job_id,
+            "sweep_id": self.sweep_id,
+            "epoch": epoch,
+            "step": step,
+            "data": data,
+        }
+        self.buffered_data.append(log_data)
+        if len(self.buffered_data) > self.max_logs_in_buffer:
+            self._flush_buffered_data()
+
+    def _training_start(self, state: State, logger: Logger) -> None:
```

Comment:
can you move this to the `init` callback? #85 will switch to expecting callbacks to be initialized on init, not training start